download/download.go

```diff
@@ -73,6 +73,16 @@ func (d *Downloader) WithLogAttrs(attrs [][2]string) *Downloader {
 	return d
 }
 
+// ClearCache resets the etag value on the downloader
+func (d *Downloader) ClearCache() {
+	d.etag = ""
+}
+
+// GetCache returns the etag value on the downloader
+func (d *Downloader) GetCache() string {
+	return d.etag
+}
+
```

Comment:
Looks like this API is only for testing, right? I'd maybe either remove it or rename to `GetETag()`.

Being a public API on the golang API's I'm learning to be much more conservative with these. If later on we changed to have the downloader have more caching mechanisms than just an etag this API would need to change (`ClearCache()` is pretty safe, it can always do whatever it needs to make the downloader clear its cache and the caller doesn't need to know anything about _how_ it was caching).

For the test cases in the bundle/discovery plugins we might want to use the http test server and just use the real mechanism to ensure we get a download the second time rather than add a public API to the downloader and check directly.

We can also easily test that in `download_test.go` where we already have the kind of test server setup, plus we can look at the private fields for the downloader to validate the etag state before/after clearing the cache (without the public API).

Reply:
Made couple of changes: 

- Removed the `GetETag()` public API. So `ClearCache()` is the only public API on the downloader
- Updated the tests in `download_test.go` to cover different etag caching scenarios

LMKWYT.